core/modules/wdb/QueryAction.cc

```diff
@@ -164,6 +167,9 @@ QueryAction::Impl::Impl(QueryActionArg const& a)
       _poisoned(false),
       _sendChannel(a.task->sendChannel),
       _user(a.task->user) {
+    char buf[MAX_NAME_LENGTH];
+    gethostname(buf, MAX_NAME_LENGTH);
+    _hostName = buf;
```

Comment:
If there is more than one instance of this class I'd avoid calling `gethostname()` repeatedly  with simple caching like:

    namespace {
        std::string const& hostname() {
            static std::string cachedHostname;
            if (cachedHostname.empty()) {
                char buf[64];
                gethostname(buf, sizeof buf);
                cachedHostname = buf;
            }
            return cachedHostname;
        }
    }